euphonic/spectra.py

```diff
@@ -103,23 +103,27 @@ def _is_bin_edge(self, bin_ax, data_ax='y'):
         bins = getattr(self, f'{bin_ax}_data')
         data = getattr(self, f'{data_ax}_data')
         if len(bins) == data.shape[enum[bin_ax]] + 1:
-            return True, bins
-        return False, bins
+            is_bin_edge = True
+        else:
+            is_bin_edge = False
+        return is_bin_edge, bins.magnitude, bins.units
 
     def _get_bin_edges(self, bin_ax='x'):
```

Comment:
I think I get what's going on here, you're passing around separate magnitudes and units internally. It would be a bit easier to follow with some type annotations! Doesn't need to be added to this PR though. Maybe I'll do it in the other spectrum work after rebasing...